src/wp-includes/class-wp-query.php

```diff
@@ -719,7 +719,8 @@ public function fill_query_vars( $array ) {
 	public function parse_query( $query = '' ) {
 		if ( ! empty( $query ) ) {
 			$this->init();
-			$this->query = $this->query_vars = wp_parse_args( $query );
+			$this->query      = wp_parse_args( $query );
```

Comment:
`$this->` stands for the class. We have not instantiated this in the class anywhere else. I might be wrong but shouldn't this be scoped in the function?

Reply:
I think $this is a special case is a accessible through the function. This function calls `init()` within the class and the 2 variables that get defined are declared public when the class is loaded so I think this is fine as it stands.

Reply:
I've been looking for a better explanation, this is the best I've found so far:

https://www.php.net/manual/en/language.oop5.properties.php

> The pseudo-variable $this is available inside any class method when that method is called from within an object context. $this is the value of the calling object.